Phase 3 funding: Addressing long-standing bugs and feature requests by monetary bounty (#276)
astropy/astropy-project: finance/proposal-calls/cycle3/bounties.md

We believe that there is no way to design a perfect process in advance, we'll just have to take the plunge and try it out. Thus, we propose to put a limited amount of money (see budget below) into "issue bounties".
We'll in particular see if the community is interested in earning the bug bounties and what amount of money is required to make this work.

To this end we will select the following issues, which are over 3 years old, look reasonably self-contained, and could be addressed (i.e. don't need work on several other issues before they can be addressed) by a variety of developers (i.e. they may be complex in programming, but don't require APEs or decision on API that only the maintainers can make):

> - issue xxx: 500 $ for PR and 125 $ for review
> - issue yyy: 1000 $ for PR and 250 $ for review
> -

Author: ```suggestion
- https://github.com/astropy/astropy/issues/4112 - 300 $ + 75 $ for review
- https://github.com/astropy/astropy/issues/3720 - 300 $+ 75 $ for review
- https://github.com/astropy/astropy/issues/4121 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/4610 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/1803 - 3000 $ + 750 $ for review
- https://github.com/astropy/astropy/issues/1783 - 1600 $  + 400 $ for review
- https://github.com/astropy/astropy/issues/3214 - 300 $ + 75 $ for review
- https://github.com/astropy/astropy/issues/3081 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/6848 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/6618 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/6617 - 600 $ + 150 $ for review
- https://github.com/astropy/astropy/issues/6583 - 3000 $ + 750 $ for review
- https://github.com/astropy/astropy/issues/6549 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/6548 - 1600 $ + 400 $ for review
```

> (please suggest issues for this list during the discussion phase)

Author: ```suggestion
(please suggest further issues for this list during the discussion phase)
```

Bounty hunters assume the risk that somebody else might have a PR to close their issue first or that an issue might turn out to be more complicated that expected. The bounty sums are set when this proposal is accepted and will not be adjusted.

We do not want this proposal to increase the workload for volunteer reviewers, so whoever leaves the most substantial review comments, can claim 25% of the bounty sum. We anticipate that it will tyically be clear and uncontroversial who has the right to claim the "review bounty" and that some fraction of reviewers might not claim the reward anyway. However, the point of this program is to test such assumptions in real live. For this year, and this year only, all disputes about who can claim the bounty will be decided the the group of people listed as organizers of this program. 

### Approximate Budget

> about 20000 $
> 
> (exact sum to be calculated beased on the list of issues with a bounty that will be collected in the discussion phase.)

Author: ```suggestion
10000 $ to 20000 $

It is to be expected that developers will selectively pick and choose and not all issues will be worked on. We will keep a running tally of the funding; this program simply ends when it runs out of money.
```

We believe that there is no way to design a perfect process in advance, we'll just have to take the plunge and try it out. Thus, we propose to put a limited amount of money (see budget below) into "issue bounties".
We'll in particular see if the community is interested in earning the bug bounties and what amount of money is required to make this work.

To this end we will select the following issues, which are over 3 years old, look reasonably self-contained, and could be addressed (i.e. don't need work on several other issues before they can be addressed) by a variety of developers (i.e. they may be complex in programming, but don't require APEs or decision on API that only the maintainers can make):

- https://github.com/astropy/astropy/issues/4112 - 300 $ + 75 $ for review
- https://github.com/astropy/astropy/issues/3720 - 300 $+ 75 $ for review
- https://github.com/astropy/astropy/issues/4121 - 1600 $ + 400 $ for review
- https://github.com/astropy/astropy/issues/4610 - 1600 $ + 400 $ for review

> - https://github.com/astropy/astropy/issues/1803 - 3000 $ + 750 $ for review

Reviewer: I haven't been following this closely. Is this the *max possible* or flat rate? Seems like a lot of money for one issue, and what if it turns out to be really simple fix? 

My comment is in general, not just about that one issue.

Author: I'm suggesting that for this particular issue. That's based on my best guess how long it might take. The point here would be to offer fixed-price tasks. If someone finds an easy fix, that's great. However, all those issue have been open for years. So, if there was an obvious, simple fix, I would assume someone would have found it by now. I looked at the issues and tried to make a realistic estimate of how long it might take. 

That's the weak point here - I may not have seen how easy that it. But that's part of this proposal, to see if that works or not. If you have a different suggestion or a better idea to come up with these numbers, please let me know.

If you are opposed to fixed-price offers in general, please downvote.

Bounty hunters assume the risk that somebody else might have a PR to close their issue first or that an issue might turn out to be more complicated that expected. The bounty sums are set when this proposal is accepted and will not be adjusted.

We do not want this proposal to increase the workload for volunteer reviewers, so whoever leaves the most substantial review comments, can claim 25% of the bounty sum. We anticipate that it will tyically be clear and uncontroversial who has the right to claim the "review bounty" and that some fraction of reviewers might not claim the reward anyway. However, the point of this program is to test such assumptions in real live. For this year, and this year only, all disputes about who can claim the bounty will be decided the the group of people listed as organizers of this program. 

### Approximate Budget

> 10000 $ to 20000 $

Author: Below 10000 $ it makes little sense to set up this program. It needs to be promoted (inside and outside of astronomy) and we want to evaluate feedback to see if this bounty program should be continued next year so we need enough funding for multiple issues.
